genesyscloud/resource_genesyscloud_widget_deployment.go

```diff
@@ -0,0 +1,405 @@
+		Name:                   &name,
+		Description:            &description,
+		AuthenticationRequired: &auth_required,
+		Disabled:               &disabled,
+		Flow:                   flowId,
+		AllowedDomains:         allowed_domains,
+		ClientType:             &client_type,
+		ClientConfig:           client_config,
+	}
+
+	log.Printf("Creating widgets deployment %s", name)
+	widget, _, err := widgetsAPI.PostWidgetsDeployments(createWidget)
+
+	if err != nil {
+		return diag.Errorf("Failed to create widget deploynment %s, %s", name, err)
+	}
+	d.SetId(*widget.Id)
+
+	return readWidgetDeployment(ctx, d, meta)
+}
+
+func deleteWidgetDeployment(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	widgetAPI := platformclientv2.NewWidgetsApiWithConfig(sdkConfig)
+
+	log.Printf("Deleting widget deployment %s", name)
+	_, err := widgetAPI.DeleteWidgetsDeployment(d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to delete widget deployment %s: %s", name, err)
+	}
+
+	time.Sleep(5 * time.Second)
+
+	return withRetries(ctx, 30*time.Second, func() *resource.RetryError {
+		_, resp, err := widgetAPI.GetWidgetsDeployment(d.Id())
+		if err != nil {
+			if isStatus404(resp) {
+				log.Printf("Widget deployment %s deleted", name)
+				return nil
+			}
+			return resource.NonRetryableError(fmt.Errorf("Error deleting widget deployment %s: %s", d.Id(), err))
+		}
+		return resource.RetryableError(fmt.Errorf("Widget deployment %s still exists", d.Id()))
+	})
+}
+
+func updateWidgetDeployment(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	auth_required := d.Get("authentication_required").(bool)
+	disabled := d.Get("disabled").(bool)
+	flowId := buildSdkDomainEntityRef(d, "flow_id")
+	allowed_domains := buildSdkAllowedDomains(d) //Need to make this an array of strings.
+	client_type := d.Get("client_type").(string)
+	client_config, client_config_err := buildSDKClientConfig(client_type, d)
+
+	if client_config_err != nil {
+		return diag.Errorf("Failed updating widget deployment %s, %s", name, client_config_err)
+	}
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	widgetsAPI := platformclientv2.NewWidgetsApiWithConfig(sdkConfig)
+
+	updateWidget := platformclientv2.Widgetdeployment{
+		Name:                   &name,
+		Description:            &description,
+		AuthenticationRequired: &auth_required,
+		Disabled:               &disabled,
+		Flow:                   flowId,
+		AllowedDomains:         allowed_domains,
+		ClientType:             &client_type,
+		ClientConfig:           client_config,
+	}
+
+	log.Printf("Updating widget deployment %s", name)
+	widget, _, err := widgetsAPI.PutWidgetsDeployment(d.Id(), updateWidget)
+
+	if err != nil {
+		return diag.Errorf("Failed to widget deployment %s, %s", name, err)
```

Comment:
Failed to widget deployment -> Failed to update widget deployment

Reply:
Fixed.